app/src/main/java/com/benefit/services/SearchService.java

```diff
@@ -13,4 +29,44 @@ public SearchService(DatabaseDriver databaseDriver) {
         this.databaseDriver = databaseDriver;
         this.productsCollection = this.databaseDriver.getCollectionByName(COLLECTION_NAME_PRODUCTS);
     }
+
+    public MutableLiveData<List<Product>> getProductsBySearchString(String searchQuery) {
```

Comment:
I think you shuold return LiveData and not MutableLiveData.

Reply:
same thing every place you use Livedata

Reply:
thanks, changed it only in methods signatures to expose a read-only object. (as explained here: https://stackoverflow.com/questions/54600660/why-does-livedata-better-than-mutablelivedata)